### PR TITLE
docs(core): update `owasp guide project` reference url to latest owasp site

### PR DESCRIPTION
**Pull request by Himanshu-Tamrakar**
owasp guide reference url is pointing to archived `owasp foundation wiki` as per the website description and may not contain the latest information.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/security.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 This topic describes Angular's built-in protections against common web-application vulnerabilities and attacks such as cross-site scripting attacks.
 It doesn't cover application-level security, such as authentication and authorization.
 
-For more information about the attacks and mitigations described below, see [OWASP Guide Project](https://www.owasp.org/index.php/Category:OWASP_Guide_Project).
+For more information about the attacks and mitigations described below, see [OWASP Guide Project](https://owasp.org/www-project-web-security-testing-guide/).
```
Comment (pkozlowski-opensource) [inline aio/content/guide/security.md:6]: I'm not sure this is the right replacement link. The one you are suggesting is the new version of https://wiki.owasp.org/index.php/OWASP_Testing_Project - and this is the different document.

Probably the most appropriate replacement link would be either https://owasp.org/www-project-security-knowledge-framework/ or https://github.com/OWASP/DevGuide but it doesn't seem to have active development for over a year.

**Comment by jessicajaniuk** [comment]
@Himanshu-Tamrakar Looks like the commit message needs to be fixed according to our linting rules. Can you amend your commit message and resolve the issue raised by @pkozlowski-opensource? Once you do that, we can review and land this PR.

**Comment by Himanshu-Tamrakar** [comment]
I have updated the commit message @[jessicajaniuk](https://github.com/jessicajaniuk).

File: aio/content/guide/security.md (hunk 1 above)
Comment (Himanshu-Tamrakar) [reply aio/content/guide/security.md:6]: We have https://wiki.owasp.org/index.php/OWASP_Testing_Project, suggested by me and,
we also https://owasp.org/www-project-security-knowledge-framework/ or https://github.com/OWASP/DevGuide suggested by @pkozlowski-opensource.

I think https://wiki.owasp.org/index.php/OWASP_Testing_Project is more appropriate as it contains https://owasp.org/www-project-web-security-testing-guide/stable/ link and It has lots of security-related content.

**Comment by jessicajaniuk** [comment]
@Himanshu-Tamrakar It looks like you updated the PR title, but not the commit message header itself. Can you run a `git commit --amend` and use the same message you put for the PR header? Once you force push that change, that should fix the lint error.

**Comment by dylhunn** [comment]
@pkozlowski-opensource Looking through our PR backlog as part of caretaking. Is the new link OK to you?

**Pull request closed without merging**